crates/red_knot_python_semantic/src/types/call/bind.rs

```diff
@@ -536,7 +536,9 @@ impl<'db> Bindings<'db> {
                     }
 
                     Some(KnownFunction::Overload) => {
-                        overload.set_return_type(todo_type!("overload[..] return type"));
+                        if let [Some(ty)] = overload.parameter_types() {
+                            overload.set_return_type(*ty);
+                        }
                     }
```

Comment:
This whole block should become a no-op once we understand the (generic identity function) definition of `typing.overload` from typeshed? (Requires legacy generics support). Might be worth a TODO to remove this later.

Reply:
Added a TODO and a small test case that validates the effect of `bar = overload(foo)`.